docs/content/en/docs/reference/clusterspec/gitops.md

```diff
@@ -8,7 +8,133 @@ description: >
+
+## Flux Configuration
+The flux configuration spec has three optional fields, regardless of the chosen git provider.
+
+### Flux Configuration Spec Details
+### __systemNamespace__ (optional)
+* __Description__: Namespace in which to install the gitops components in your cluster. Defaults to `flux-system`
+* __Type__: string
+
+### __clusterConfigPath__ (optional)
+
+* __Description__: The path relative to the root of the git repository where EKS Anywhere will store the cluster configuration files. Defaults to the cluster name
+* __Type__: string
+
+### __branch__ (optional)
+
+* __Description__: The branch to use when committing the configuration. Defaults to `main`
+* __Type__: string
+
+EKS Anywhere currently supports two git providers for FluxConfig: Github and Git.
+
+## Github provider
+Please note that for the Flux config to work successfully with the Github provider, the environment variable `EKSA_GITHUB_TOKEN` needs to be set with a valid [GitHub PAT](https://github.com/settings/tokens/new).
+This is a generic template with detailed descriptions below for reference:
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: my-cluster-name
+spec:
+  ...
+  #GitOps Support
+  gitOpsRef:
+    name: my-github-flux-provider
+    kind: FluxConfig
+---
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: FluxConfig
+metadata:
+  name: my-github-flux-provider
+spec:
+  systemNamespace: "my-alternative-flux-system-namespace"
+  clusterConfigPath: "path-to-my-clusters-config"
+  branch: "main"
+  github:
+    personal: true
+    repository: myClusterGitopsRepo
+    owner: myGithubUsername
+
+---
+```
+
+### github Configuration Spec Details
+### __repository__ (required)
+
+* __Description__: The name of the repository where we will store your cluster configuration, and sync it to the cluster. If the repository exists, we will clone it from the git provider; if it does not exist, we will create it for you.
+* __Type__: string
+
+### __owner__ (required)
+
+* __Description__: The owner of the git repository; either a github username or github organization name. The Personal Access Token used must belong to the owner if this is a personal repository, or have permissions over the organization if this is not a personal repository.
+* __Type__: string
+
+### __personal__ (optional)
+
+* __Description__: Is the repository a personal or organization repository? If personal, this value is true; otherwise, false. If using an organizational repository (e.g. personal is false) the owner field will be used as the organization when authenticating to github.com (http://github.com/)
+* __Default__: true
+* __Type__: boolean
+
+## Git provider
+
+Before you create a cluster using the Git provider, you will need to set and export these environment variables.
+
+`EKSA_GIT_KNOWN_HOSTS`: Path to your known hosts file
+
+*-** TODO: insert info on how to obtain known hosts info for the git provider*
```

Comment:
EKS Anywhere uses the provided known hosts file to verify the identity of the git provider when connecting to it with SSH. The `EKSA_GIT_KNOWN_HOSTS` environment variable should be a path to a known hosts file containing entries for the git server to which you'll be connecting.

For example, if you wanted to provide a known hosts file which allows you to connect to and verify the identity of github.com using private key using the key algorithm`ecdsa` , you can use the OpenSSH utility [`ssh-keyscan`](https://manpages.ubuntu.com/manpages/xenial/man1/ssh-keyscan.1.html) to obtain the known host entry used by github.com for the `ecdsa` key type. EKS Anywhere supports `ecdsa`, `rsa`, and  `ed25519` key types, which can be specified via the `sshKeyAlgorithm` field of the `git` provider config.

```bash
ssk-keyscan -t ecdsa github.com >> my_eksa_known_hosts
```

This will produce a file which contains known-hosts entries for the `ecdsa` key type supported by github.com, mapping the host to the key-type and public key.

```
github.com ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEmKSENjQEezOmxkZMy7opKgwFB9nkt5YRrYMjNuG5N87uRgg6CLrbo5wAdT/y6v0mKV0U2w0WZ2YB/++Tpockg=
```

EKS Anywhere will use the content of the file at the path `EKA_GIT_KNOWN_HOSTS` to verify the identity of the remote git server, and the provided known hosts file must contain an entry for the remote host and key type.